### PR TITLE
remove not needed initializes in closets

### LINKED ISSUE #22933: Closets call initialize() twice under certain circumstances
First on https://github.com/vgstation-coders/vgstation13/blob/Bleeding-Edge/code/game/objects/structures/crates_lockers/closets.dm#L41

And then on https://github.com/vgstation-coders/vgstation13/blob/Bleeding-Edge/code/game/objects/structures/crates_lockers/closets.dm#L74

There are no checks in place to prevent the second call, resulting in duplicate items.

[easy][bugfix]

### PR DESCRIPTION
**Pull request by jjpark-kb**
closes: #22933
:cl:
 * tweak: lockers in vaults and away missions will no longer contain twice the amount of things they're supposed to.

### CODE REVIEW

**Comment by PrimeDSS13** [comment]
sleep tight double HoS locker loot

**Comment by SonixApache** [comment]
this ain't what's causing that, @PrimeDSS13

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 1]
```diff
@@ -51,8 +51,7 @@
 	for(var/path in to_spawn)
 		var/amount = to_spawn[path] || 1
 		for(var/i in 1 to amount)
-			var/atom/A = new path(src)
-			A.initialize()  // recursive
```
Comment (DamianX) [inline code/game/objects/structures/crates_lockers/closets.dm]: Why did you remove this? iirc this is needed for headsets inside lockers to be properly initialized

Comment (jjpark-kb) [reply code/game/objects/structures/crates_lockers/closets.dm]: I did not know that this was needed for headsets, let me bring it back
